src/sonar_widget/SonarPlot.cc

```diff
@@ -161,7 +171,7 @@ void SonarPlot::drawOverlay()
             base::Angle ang = lastSonar.bearings[((lastSonar.beam_count - 1) * 1.0 / 4) * (i-1)];
             QPoint point(origin.rx() + BINS_REF_SIZE * sin(ang.rad) * scaleX, origin.ry() - BINS_REF_SIZE * cos(ang.rad) * scaleY);
             painter.drawLine(origin, point);
-            str.setNum(ang.getDeg());
+            str.setNum(roundf(ang.getDeg() * 10) / 10);
```

Comment:
`roundf` is for floats `ang.getDeg()` is a double.

In any case, better use the formatting arguments of [`QString::arg()`](http://doc.qt.io/qt-4.8/qstring.html#arg-20) to format the number instead of doing some rounding mumbo-jumbo.... These are floating-point values, there's no guarantee that what you compute won't be equal to "-18.99999999"